src/ORM/Connect/MySQLSchemaManager.php

```diff
@@ -647,4 +682,143 @@ protected function defaultClause($values)
+
+        // If neither spec is generated, we can just use ALTER
+        if (!$existingSpecType && !$newSpecType) {
+            return false;
+        }
+
+        // Virtual generated columns cannot be altered to stored generated columns
+        // Virtual generated columns cannot be altered to nongenerated columns
+        if ($existingSpecType === DBGeneratedColumn::GENERATION_VIRTUAL) {
+            if ($newSpecType !== DBGeneratedColumn::GENERATION_VIRTUAL) {
+                return true;
+            }
+            return false;
+        }
+
+        // Stored generated columns cannot be altered to virtual generated columns
+        // Stored generated columns can be altered to nongenerated columns
+        if ($existingSpecType === DBGeneratedColumn::GENERATION_STORED) {
+            if ($newSpecType === DBGeneratedColumn::GENERATION_VIRTUAL) {
+                return true;
+            }
+            return false;
+        }
+
+        // Nongenerated columns can be altered to stored but not virtual generated columns
+        return $newSpecType === DBGeneratedColumn::GENERATION_VIRTUAL;
+    }
+
+    /**
+     * Get what type of storage is used for a generated column, or null if not a generated column.
+     */
+    private function getGeneratedType(string $spec): ?string
+    {
+        // If it ends with "STORED" or "VIRTUAL", it's a generated column spec.
+        preg_match('/(STORED|VIRTUAL)$/i', $spec, $matches);
+        return strtoupper($matches[1] ?? '');
+    }
+
+    /**
+     * Normalises a MySQL generated column expression.
+     * This is needed to robustly validate whether the expression we want matches the expression
+     * that's already in the database, because MySQL sometimes makes changes internally that we
+     * need to normalise out, such as:
+     * - escaping quotes
+     * - double-escaping backslashes
+     * - adding the charset before explicit strings
+     * - changing case of keywords
+     * - using backticks instead of ANSI quotes around columns names
+     * - adding spaces around operators
+     */
+    private function normaliseGeneratedColumnExpression(string $expression): string
+    {
+        // Clean up string literals. Easier to do here than in the AST because the
+        // escaped quotes MySQL gives us change the representation of the parsed string.
+        $expression = preg_replace_callback(
+            // Both \' and '' are valid escape sequences for a single quote in a string literal
+            // see https://dev.mysql.com/doc/refman/8.4/en/string-literals.html
+            "/\\\\'(?:[^']|''|\\\\\\\\')*\\\\'/",
+            function ($matches) {
+                // Replace escaped quotes around the string literal with single unescaped quotes
+                $x = preg_replace("/^\\\\'(.*)\\\\'$/", "'$1'", $matches[0]);
+                // Tidy up escaped backslashes, e.g. for FQCN
+                return str_replace('\\\\\\\\', '\\\\', $x);
```

Comment:
When we pass in `SilverStripe\\Assets\\Folder` as a string as part of an expression, then ask MySQL what the expression is, it gives us back `SilverStripe\\\\Assets\\\\Folder`